uamqp/client.py

```diff
@@ -373,12 +373,25 @@ def _transfer_message(self, message_delivery, timeout=0):
     def _on_send_complete(self, message_delivery, message, reason, state):
         # TODO: check whether the callback would be called in case of message expiry or link going down
         # and if so handle the state in the callback
-        if SEND_DISPOSITION_ACCEPT in state:
+        if state and SEND_DISPOSITION_ACCEPT in state:
             message_delivery.state = MessageDeliveryState.Ok
         else:
             # TODO: sending disposition state could only be rejected/accepted?
-            message_delivery.state = MessageDeliveryState.Error
-            message_delivery.reason = reason
+            # TODO: error action
+            if not state and reason == LinkDeliverySettleReason.NotDelivered:
+                # TODO: message is not delivered
+                message_delivery.state = MessageDeliveryState.Error
```

Comment:
Shouldn't we add the reason before returning?
`message_delivery.reason = reason`
While 'NotDelivered' is not hugely useful - it's still good to know that it didn't fail _after_ delivery.

Reply:
updated ^^